diff --git a/pjsip/include/pjsua2/media.hpp b/pjsip/include/pjsua2/media.hpp
index b6749d1aa..e1cb5eef9 100644
--- a/pjsip/include/pjsua2/media.hpp
+++ b/pjsip/include/pjsua2/media.hpp
@@ -2925,6 +2925,15 @@ struct AudDevErrorEvent
     pj_status_t             status;     /**< The status code.       */
 };
 
+/**
+ * This structure describes the rtcp fb event.
+ */
+struct RtcpFbEvent
+{
+    pjmedia_rtcp_fb_type fbType;
+    bool isParamLengthZero;
+};
+
 /**
  * Media event data.
  */
@@ -2938,6 +2947,11 @@ typedef union MediaEventData {
      * Audio device error event data.
      */
     AudDevErrorEvent        audDevError;
+
+    /**
+     * Rtcp FB event data.
+     */
+    RtcpFbEvent     rtcpFb;
     
     /**
      * Pointer to storage to user event data, if it's outside
diff --git a/pjsip/src/pjsua2/media.cpp b/pjsip/src/pjsua2/media.cpp
index 0710ed267..3987c1a73 100644
--- a/pjsip/src/pjsua2/media.cpp
+++ b/pjsip/src/pjsua2/media.cpp
@@ -2355,6 +2355,9 @@ void MediaEvent::fromPj(const pjmedia_event &ev)
         data.audDevError.dir = ev.data.aud_dev_err.dir;
         data.audDevError.id = ev.data.aud_dev_err.id;
         data.audDevError.status = ev.data.aud_dev_err.status;
+    } else if (type == PJMEDIA_EVENT_RX_RTCP_FB) {
+        data.rtcpFb.fbType = ev.data.rx_rtcp_fb.cap.type;
+        data.rtcpFb.isParamLengthZero = ev.data.rx_rtcp_fb.cap.param.slen == 0;
     }
     pjMediaEvent = (void *)&ev;
 }
